solidity/dashboard/src/App.js

```diff
@@ -26,6 +27,32 @@ const App = () => (
 )
 
 const AppLayout = () => {
+  const { isConnected, connector, yourAddress } = useWeb3Context()
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    const accountChangedHandler = (address) => {
+      dispatch({ type: "app/account_changed", payload: { address } })
+    }
+
+    const disconnectHandler = () => {
+      dispatch({ type: "app/logout" })
+    }
+
+    if (isConnected && connector) {
+      dispatch({ type: "app/login", payload: { address: yourAddress } })
+      connector.on("accountsChanged", accountChangedHandler)
+      connector.once("disconnect", disconnectHandler)
+    }
+
+    return () => {
+      if (connector) {
+        connector.removeListener("accountsChanged", accountChangedHandler)
+        connector.removeListener("disconnect", disconnectHandler)
+      }
+    }
+  }, [isConnected, connector, dispatch, yourAddress])
```

Comment:
The main idea was to avoid connecting the `Web3Context` with redux. The `Web3Context` can exist without third-party components and can handle events in specific ways that only make sense inside `Web3Context`. One of the disadvantages of the proposed solution is that if, eg., we put `Web3Context` over the Redux provider in the app tree, `Web3Context` will not work.